src/node.cc

```diff
@@ -3741,7 +3750,7 @@ static void ParseArgs(int* argc,
 #endif /* HAVE_OPENSSL */
 #if defined(NODE_HAVE_I18N_SUPPORT)
     } else if (strncmp(arg, "--icu-data-dir=", 15) == 0) {
-      icu_data_dir = arg + 15;
+      icu_data_dir.assign(arg, 15);
```

Comment:
When converting `openssl_config` to std::string, I discovered this copies the first 15 chars of arg into target, it needs to be `icu_data_dir.assign(arg + 15);` or something of the like. @srl295 Since tests pass, this means that this command line option has no test coverage.